opsmngr/backup_configs.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	backupConfigsBasePath = "groups/%s/backupConfigs"
+)
+
+// BackupConfigsService is an interface for using the Backup Configurations
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/backup-configurations/
+type BackupConfigsService interface {
+	List(context.Context, string, *atlas.ListOptions) (*BackupConfigs, *atlas.Response, error)
+	Get(context.Context, string, string) (*BackupConfig, *atlas.Response, error)
+	Update(context.Context, string, string, *BackupConfig) (*BackupConfig, *atlas.Response, error)
+}
+
+// BackupConfigsServiceOp provides an implementation of the BackupConfigsService interface
+type BackupConfigsServiceOp service
+
+var _ BackupConfigsService = &BackupConfigsServiceOp{}
+
+// BackupConfigs represents a Backup configuration in the MongoDB Ops Manager API
+type BackupConfig struct {
+	GroupID            string   `json:"groupId,omitempty"`
+	ClusterID          string   `json:"clusterId,omitempty"`
+	StatusName         string   `json:"statusName,omitempty"`
+	StorageEngineName  string   `json:"storageEngineName,omitempty"`
+	EncryptionEnabled  bool     `json:"encryptionEnabled,omitempty"`
+	SSLEnabled         bool     `json:"sslEnabled,omitempty"`
```

Comment:
`omitempty` and bool should use a pointer on the bool or `false` won't be serialised